### DISCUSSION TITLE
Ability to disable the avatar/user menu on registration flow

### DISCUSSION THREAD

**Discussion by stursby** [filamentphp/filament#12596] [Help] [answered]
### Package

filament/filament

### Package Version

v3.2.72

### Laravel Version

v11.6.0

### Livewire Version

v3.4.12

### PHP Version

PHP 8.3.4

### Problem description

Not sure if this is really a bug... but when I enable registration and emailVerification, the avatar/user menu shows up, which feels odd since they aren't really a user yet?

```php
// /app/Providers/Filament/AdminPanelProvider.php

return $panel
      ->default()
      ->id('admin')
      ->path('admin')
      ->login()
      ->registration() // added
      ->emailVerification() // added
```

I'm wondering if there's an easy way to just disable it during registration.

<img width="1326" alt="image" src="https://github.com/filamentphp/filament/assets/527849/9d3a15d6-7267-4f57-a59b-2a5d9f6dfd2e">

### Expected behavior

I would expect that you can't see you avatar, profile link (the actual profile page doesn't work until email confirmed), etc until after the registration process.

### Steps to reproduce

Setup repo, register for new account, and on the email verification page you'll see the little user avatar / menu in top right. (also make sure to run `php artisan queue:work` if you want to actually receive the email)

### Reproduction repository

https://github.com/stursby/filament-avatar-issue

**Comment by xalabama** [comment]
Hmm, i think it depends of what you want the user can do if he has not verified his email address. I think it make sense to allow the user to change his profile information. Some other applications i know want to keep the user inside the application and show all the features, so the user only needs to verify his email address to not get removed from the system. However I think it's not a bug.

**Reply by stursby** [reply]
Yea, I don't really think it's bug either and I guess it's just personal preference. I would prefer that stay hidden until the user actually completes the registration process 🤷‍♂️

**Comment by stursby** [comment, marked as answer]
In Discord @saade provided a solution by extending the default `EmailVerificationPrompt` class.

So, for example:

1. Make a new class `EmailVerification.php` (I put it in `app/Pages/EmailVerification.php`

```php

<?php

namespace App\Filament\Pages;

use Filament\Pages\Auth\EmailVerification\EmailVerificationPrompt;

class EmailVerification extends EmailVerificationPrompt
{
    protected bool $hasTopbar = false;
}
```

2. Update your panel provider

```php

// import the new custom class
use App\Filament\Pages\EmailVerification;

...

public function panel(Panel $panel): Panel
    {
        return $panel
            ->default()
            ->id('admin')
            ->path('admin')
            ->login()
            ->registration()
            ->passwordReset()
            // ->emailVerification() // replace this with line below
            ->emailVerification(EmailVerification::class)
```

**Comment by MACscr** [comment]
Isnt this just a workaround though and should already be the default?
